internal/core/model.rs

```diff
@@ -119,11 +119,13 @@ impl ModelTracker for () {
 /// }
 /// ```
 pub trait Model {
-    /// The model data: A model is a set of row and each row has this data
+    /// The model data: A model is a set of rows and each row has this data
     type Data;
-    /// The amount of row in the model
+    /// The number of rows in the model
     fn row_count(&self) -> usize;
-    /// Returns the data for a particular row. This function should be called with `row < row_count()`.
+    /// Returns the data for a particular row.
```

Comment:
Just a sanity question: Is the C++ view affected similarly with the std::optional?

Reply:
Yes, kinda. It's worse because it's UB. But one shouldn't return a null optional from row_data anyway.
But you're right that it should also be fixed even if C++ devs are used to random crash when they do something wrong :fearful: